src/content/deployment/flavors-ios.md

```diff
@@ -0,0 +1,475 @@
+short-title: Flavors (iOS and macOS)
+description: >
+  How to create Flutter flavors for an iOS or macOS app.
+---
+
+This guide shows you how to create Flutter flavors for an
+iOS or macOS app.
+
+## Overview
+
+A Flutter flavor is basically a collection of settings that
+define how a specific version of your app should be built
+and run. For example, a flavor could determine which icon,
+app name, API key, feature flag, and logging level is
+associated with a specific version of your app.
+
+If you want to create Flutter flavors for an iOS app, you'll
+need to do so in Xcode. Xcode does not have a concept called
+"flavor". Instead, you'll need to set up something called a
+scheme and attach custom configurations to it.
+
+The following illustrates an example of two Flutter flavors
+(staging, production) as Xcode schemes with custom Xcode
+configurations assigned to them:
+
+<table class="table table-striped">
+  <thead>
+    <tr>
+      <th>Scheme</th>
+      <th>Configurations for the scheme</th>
+    </tr>
+  </thead>
+  <tbody>
+    <tr>
+      <td>staging</td>
+      <td>
+        Debug-staging</br>
+        Profile-staging</br>
+        Release-staging</br>
+      </td>
+    </tr>
+    <tr>
+      <td>production</td>
+      <td>
+        Debug-production</br>
+        Profile-production</br>
+        Release-production</br>
+      </td>
+    </tr>
+  </tbody>
+</table>
+
+## Configure Xcode schemes
+
+The following steps show how to configure two Xcode schemes
+called `staging` and `production` for your Flutter iOS
+project. You can also use these steps to set up a
+macOS project by replacing any reference to `iOS`
+with `macOS`.
+
+For a seamless workflow, we've started with a
+new Flutter project called `flavors_example`, but you can
+always start with an existing project.
+
+1.  Create a new Flutter project called `flavors_example`.
+
+    ```console title="console"
+    $ flutter create flavors_example
+    ```
+
+1.  Open the default Xcode workspace for the iOS version of
+    the `flavors_example` project.
+
+    ```console title="console"
+    $ cd flavors_example && open ios/Runner.xcworkspace
+    ```
+
+1.  Open the `flavors_example` project in the
+    Xcode project navigator:
+
+    * Open the **project navigator**
+      (**View** > **Navigators** > **Project**).
+    
+    * In the **project navigator**, at the top, select
+      **Runner**.
+
+1.  Create schemes in Xcode:
+
+    * Open the **New Scheme** window
+      (**Product > Scheme > New Scheme**).
+
+    * In the **Name** box, enter `staging`.
+
+    * Click **Okay** to add the new scheme.
+
+    * Repeat the previous steps for a scheme called
+      `production`.
+
+    * When finished, check to make sure that you
+      have the following schemes:
+
+      ![Schemes for Flutter flavors](/assets/images/docs/flavors/flavors-ios-schemes.png){:width="100%"}
+
+    :::note
+    By default, the target for the scheme is `Runner`
```

Comment:
I don't know that the default is always Runner. I would add a step to the instructions to make sure that Runner is selected as the target.

Reply:
That's true when creating a brand new project, but if they have an existing project they're adding flavors to (which I think is more likely), it may not be the case. For example, if the project has Swift Package Manager enabled, the FlutterGeneratedPluginSwiftPackage is the default.

![Screenshot 2025-02-13 at 12 41 42 PM](https://github.com/user-attachments/assets/5309abee-e30e-48d4-982c-0b21e3086f31)

This instruction isn't in our current documentation for flavors either so it can't be worse than our current docs lol, just wanted to bring it up.

I'll leave it up to your discretion since there's also a note about it.

Reply:
Thank you for the context, Victoria. I think this background strengthens the case for adding a step for Target into the docs.